src/kv/kv_types.h

```diff
@@ -136,6 +136,13 @@ namespace kv
     virtual void append(const uint8_t* replicated, size_t replicated_size) = 0;
     virtual bool verify(Term* term = nullptr) = 0;
     virtual void emit_signature() = 0;
+    virtual crypto::Sha256Hash get_replicated_state_root() = 0;
```

Comment:
Unrelated note: we will need to get rid of the dependency on `crypto` at some point.